### PR TITLE
Chore/optimize selection code

### PR DESCRIPTION
**Pull request by JimmFly**
Optimized for code readability and maintainability

### CODE REVIEW

**Comment by doodlewind** [comment]
Could you elaborate what has been reached through the optimization? Thanks!

File: packages/blocks/src/__internal__/utils/selection.ts [hunk 1]
```diff
@@ -22,6 +22,99 @@ const notStrictCharacterReg = /[^\p{Alpha}\p{M}\p{Nd}\p{Pc}\p{Join_C}]/u;
 const notStrictCharacterAndSpaceReg =
   /[^\p{Alpha}\p{M}\p{Nd}\p{Pc}\p{Join_C}\s]/u;
 
+// function fixCurrentRangeToText(
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/utils/selection.ts:25]: We shouldn't checkin large batch of commented code in the codebase (unless it's a WIP PR), it won't be maintained by others and will be staled.

File: packages/blocks/src/__internal__/utils/selection.ts [hunk 2]
```diff
@@ -49,43 +142,16 @@ function fixCurrentRangeToText(
...
-              newRange.setEnd(firstTextNode, 0);
-            }
-          }
-          range.setStart(newRange.endContainer, newRange.endOffset);
-        }
-      }
+      const rect = text.getBoundingClientRect();
+      const NewY = isForward ? rect.bottom - 6 : rect.top + 6;
+      newRange = caretRangeFromPoint(x, NewY);
+      isForward
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/utils/selection.ts:148]: What is the `null` used for here? Plain if-else looks fine here. Is this what you want to express?

``` ts
if (isForward && newRange) {
  forwardSelect(newRange, range);
}
else if (!isForward && newRange) {
  backwardSelect(newRange, range);
}
```

**Comment by JimmFly** [comment]
> Could you elaborate what has been reached through the optimization? Thanks!

I modified the code where the if...else nested too deeply and extracted some new functionality to make the code easier to read and maintain. Before that, the deepest layer of the code has 7 layers of nesting, and each function has nearly 100 lines of code at the same time, which makes me unable to understand.

**Comment by doodlewind** [comment]
Thanks for the refactoring, hope we can have a more robust selection feature!

**Pull request merged**